src/views/Settings.vue

```diff
@@ -88,14 +88,13 @@ export default defineComponent({
   methods: {
     setEComStore(store: any) {
       if(this.userProfile) {
-        this.store.dispatch('user/setEcomStore', {
-          'eComStore': this.userProfile.stores.find((str: any) => str.productStoreId == store['detail'].value)
-        })
+        const productStore = this.userProfile.stores.find((str: any) => str.productStoreId === store['detail'].value);
```

Comment:
We should not find the product store here, we could just pass the productStoreId here and action should handle if id is provided it finds the product store & set it and if product store is provided it should set it

Reply:
Sure, sir, understood.